packages/vite/src/node/server/middlewares/indexHtml.ts

```diff
@@ -115,12 +115,17 @@ const devHtmlHook: IndexHtmlTransformHook = async (
         addToHTMLProxyCache(config, url, scriptModuleIndex, contents)
 
         // inline js module. convert to src="proxy"
+        const modulePath = `${
+          config.base + htmlPath.slice(1)
+        }?html-proxy&index=${scriptModuleIndex}.js`
+
+        // invalidate the module so the newly cached contents will be served
+        server?.moduleGraph.invalidateId(config.root + modulePath)
```

Comment:
I think we should avoid extending the `moduleGraph` API at this point (this is a public API, that would count as a feature, but I'll bring this to a team meeting once we get over the backlog, I also think that more utility methods would be good here)

```suggestion
        const module = server?.moduleGraph.getModuleById(config.root + modulePath)
        if( module ) {
          server?.moduleGraph.invalidateModule(module)
        }
```

I think this is simple enough to merge without a test, but it would be great to add one (here or in another PR to avoid regressions later)